doc/sphinx/language/core/records.rst

```diff
@@ -15,188 +15,226 @@ synonym :cmd:`Structure`.
+   projection function associated with the field.
+   Glossing over some syntactic details, the two forms are:
 
-   Note here that the fields ``Rat_bottom_cond`` depends on the field ``bottom``
-   and ``Rat_irred_cond`` depends on both ``top`` and ``bottom``.
+   - :n:`@qualid__record.(@qualid__field {* @arg })` (projective) and
 
-.. example::
+   - :n:`@qualid__field {* @arg } @qualid__record` (applicative)
 
-   Let us define the rational :math:`1/2` using either the applicative or record syntax:
+   where the :n:`@arg`\s are the parameters of the inductive type.
 
-    .. coqtop:: in
+   Since the projected object is part of the notation, it is always
+   considered an explicit argument of :token:`qualid`, even if it is
+   formally declared as implicit (see :ref:`ImplicitArguments`).
 
-       Theorem one_two_irred : forall x y z:nat, x * y = 1 /\ x * z = 2 -> x = 1.
-       Admitted.
+   .. todo PR
+      "Record foo2:Prop := { x:Type }." gives the output "foo2 is defined.
+      x cannot be defined because it is informative and foo2 is not."
+      What does this mean?
 
-       (* Applicative form: values for fields should be given in order *)
-       Definition half := mkRat true 1 2 (O_S 1) one_two_irred.
+   .. example::
 
-       (* Record form: omitting inferrable fields top and bottom *)
-       Definition half' :=
-         {| sign := true;
-            Rat_bottom_cond := O_S 1;
-            Rat_irred_cond := one_two_irred |}.
+      .. todo PR: maybe make this an example of matching?
 
-.. example::
+      Let us define a function by pattern matching over a record:
 
-   Let us define a function by pattern matching over a record:
+      .. coqtop:: all
 
-   .. coqtop:: all
+         Eval compute in (
+           match half with
+           | {| negative := false; top := n |} => n
+           | _ => 0
+           end).
 
-      Eval compute in (
-        match half with
-        | {| sign := true; top := n |} => n
-        | _ => 0
-        end).
+   .. example:: Accessing record fields
 
-.. example::
+      Let us project fields of a record, using either the applicative or projection syntax:
 
-   Let us project fields of a record, using either the applicative or projection syntax:
+      .. coqtop:: all
 
-   .. coqtop:: all
+         (* projection form *)
+         Eval compute in half.(top).
 
-      Eval compute in top half.
-      Eval compute in bottom half.
-      Eval compute in Rat_bottom_cond half.
-      Eval compute in half.(top).
+         (* applicative form *)
+         Eval compute in top half.
+         Eval compute in bottom half.
+         Eval compute in Rat_bottom_nonzero half.
 
-The following settings let you control the display format for types:
+The following settings let you control the display format for record types:
 
 .. flag:: Printing Records
 
    When this :term:`flag` is on (this is the default),
    use the record syntax (shown above) as the default display format.
 
```

Comment:
I think we need a short description of primitive projections (what is it, what's is used for).  And it sounds like it's experimental (or at least will evolve), so we should say that.  Would you care to suggest some wording?

Reply:
There is a section at the end of the chapter on primitive projections. I would say that until we reach a consensus on the form we want to give to primitive projections, it is not worth to change the current documentation.

Reply:
Can we reasonably say that a feature used since almost 10 years that it is still experimental? Using "evolving" seems a good idea though, something like "note that the design of primitive projections is still liable to evolve"??

Reply:
10 years seems a very long time to reach consensus!  I'll add a sentence about evolving.

Reply:
This is a long time but it also reflects the time needed to understand the feature. At the very beginning it was motivated by an optimization of the representation of projections in memory. This interferred with implementing an emerging logical concept at this time called negative types, leading to two conceptually kinds of records (defined by a constructor, as in Variant, or by primitive destructors). Then, at some time, it was thought that one-constructor Variant could be emulated on top of primitive destructors, which was a mistake. Reverting this last step is not fully done. Etc.